src/Messages/OWSMessageSender.m

```diff
@@ -42,10 +42,12 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-#define SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(major, minor)                                                          \
-    ([[NSProcessInfo processInfo]                                                                                      \
-        isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){                                                    \
-                                            .majorVersion = major, .minorVersion = minor, .patchVersion = 0 }])
+void AssertIsOnSendingQueue()
+{
+    if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(10, 0)) {
+        dispatch_assert_queue([OWSDispatch sendingQueue]);
```

Comment:
> it might lead to spurious asserts in development

My understanding is there will be no spurious asserts. 

> my impression was that dispatch_assert_queue() only applies in debug builds

That was my assumption, but actually turns out the failed asserts will crash. I've added debug checks at https://github.com/WhisperSystems/SignalServiceKit/pull/163 and https://github.com/WhisperSystems/SignalProtocolKit/pull/19